### ISSUE TITLE
Long.parseLong, Double.parseDouble, Integer.parseInt are too lenient

### ISSUE THREAD

**Issue by dankurka** [gwtproject/gwt#16]
Originally reported on Google Code with ID 7
```
GWT Release: 1.1.10

Detailed description:
The Java implementation of Long.parseLong() throws an exception if the
format is not correct:

 "16316" OK
 "65131asd" exception
 "16dasddsad" exception
 "a1" exception

However, GWT's Long.parseLong() is implemented in JavaScript and performs
as follows

"16316" OK
 "65131asd" OK returns 65131
 "16dasddsad" OK returns 16
 "a1" exception

This is also true for Double.parseDouble, Integer.parseInt, etc.

Links to the relevant discussion group topics (optional):
http://groups.google.com/group/Google-Web-Toolkit/browse_thread/thread/c31d2f239820686c
```

Reported by `gwt.team.scottb` on 2006-09-15 18:54:45

**Comment by dankurka** [comment]
Looking at test cases we have in emul this seems to be solved

**Comment by tbroyer** [comment]
FWIW, I believe that was fixed in b87cd0664244154965b264fe06d5da4a5a0f1316 (apparently shipped in 1.4.10 or something like that (we don't have all tags for those very old releases)); there's been a few commits afterwards to actually loosen the validation: 884c1891fc1fc61c1c6307149c850711285b34e3, efa7a103ae76670701277609ace38caadda26516, b96f11ab99d84184edccc576571b1b77e6d83b2e, 2a1d511ecfe08308ecd165a9cae39d46c3b258f0; or fix edge cases (e.g. 625967fca0229876a891f582b9c25e143ab180da)
